graphql/schemabuilder/helpers.go

```diff
@@ -0,0 +1,392 @@
+package schemabuilder
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+
+	"github.com/samsarahq/thunder/graphql"
+)
+
+type Connection struct {
+	Edges    []Edge
+	PageInfo PageInfo
+}
+
+type PageInfo struct {
+	HasNextPage bool
+	EndCursor   string
+	HasPrevPage bool
+	StartCursor string
+	Pages       []string
+}
+type Edge struct {
+	Node   Node
+	Cursor string
+}
+
+type Node interface {
+	GetCursor() string
+}
+
+type ConnectionArgs interface {
+	GetFirst() int64
+	GetAfter() *string
+	GetBefore() *string
+}
+
+func (sb *schemaBuilder) constructEdgeType(typ reflect.Type) (graphql.Type, error) {
+
+	nodeType, _ := sb.getType(typ)
+
+	fieldMap := make(map[string]*graphql.Field)
+
+	nodeField := &graphql.Field{
+		Resolve: func(ctx context.Context, source, args interface{}, selectionSet *graphql.SelectionSet) (interface{}, error) {
+			if value, ok := source.(Edge); ok {
+				return value.Node, nil
+			}
+
+			return nil, fmt.Errorf("error resolving node in edge")
+
+		},
+		Type:           nodeType,
+		ParseArguments: nilParseArguments,
+	}
+	fieldMap[makeGraphql("Node")] = nodeField
+
+	cursorType, err := sb.getType(reflect.TypeOf(string("a string")))
+	if err != nil {
+		return nil, err
+	}
+
+	cursorField := &graphql.Field{
+		Resolve: func(ctx context.Context, source, args interface{}, selectionSet *graphql.SelectionSet) (interface{}, error) {
+			if value, ok := source.(Edge); ok {
+				return value.Cursor, nil
+			}
+			return nil, fmt.Errorf("error resolving cursor in edge")
+		},
+		Type:           cursorType,
+		ParseArguments: nilParseArguments,
+	}
+
+	fieldMap[makeGraphql("Cursor")] = cursorField
+
+	return &graphql.NonNull{
+		Type: &graphql.Object{
+			Name:        "Edge",
+			Description: "",
+			Fields:      fieldMap,
+		},
+	}, nil
+
+}
+
+func (sb *schemaBuilder) constructConnType(typ reflect.Type) (graphql.Type, error) {
+
+	fieldMap := make(map[string]*graphql.Field)
+
+	edgeType, err := sb.constructEdgeType(typ)
+	if err != nil {
+		return nil, err
+	}
+
+	edgesSliceType := &graphql.NonNull{Type: &graphql.List{Type: edgeType}}
+
+	edgesSliceField := &graphql.Field{
+		Resolve: func(ctx context.Context, source, args interface{}, selectionSet *graphql.SelectionSet) (interface{}, error) {
+			if value, ok := source.(Connection); ok {
+				return value.Edges, nil
+			}
+			return nil, fmt.Errorf("error resolving edges in connection")
+		},
+		Type:           edgesSliceType,
+		ParseArguments: nilParseArguments,
+	}
+
+	fieldMap[makeGraphql("Edges")] = edgesSliceField
```

Comment:
think this helper only lowercases, so you could put in the strings like that directly.